cli/cmd/inject_test.go

```diff
@@ -0,0 +1,45 @@
+package cmd
+
+import (
+	"bufio"
+	"bytes"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/sergi/go-diff/diffmatchpatch"
+)
+
+func TestInjectYAML(t *testing.T) {
+	t.Run("Run successful conduit inject on valid k8s yaml", func(t *testing.T) {
+		file, err := os.Open("testdata/inject_emojivoto_deployment.input.yml")
+		if err != nil {
+			t.Errorf("error opening test file: %v\n", err)
+		}
+
+		goldenFileBytes, err := ioutil.ReadFile("testdata/inject_emojivoto_deployment.golden.yml")
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+		expectedContent := string(goldenFileBytes)
+
+		read := bufio.NewReader(file)
+
+		output := new(bytes.Buffer)
+
+		err = InjectYAML(read, output)
+		if err != nil {
+			t.Errorf("Unexpected error injecting YAML: %v\n", err)
+		}
+
+		actualOutput := output.String()
+
+		if actualOutput != expectedContent {
```

Comment:
The logic for reading a golden file (lines 20-23) and this code for comparing the expected results equal the actual results and outputing the diff and failing if they don't, is all exactly the same between the various tests. We should combine this shared logic into a helper function that all such tests use, rather than copy-pasting it into each test.